compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1075,6 +1075,12 @@ TR::Register *OMR::X86::TreeEvaluator::integerStoreEvaluator(TR::Node *node, TR:
    if (instr && node->getOpCode().isIndirect())
       cg->setImplicitExceptionPoint(instr);
 
+   if (node->getSymbol()->isUnsafeShadowSymbol())
```

Comment:
integerStoreEvaluator handles all kinds of integer data types, including Int8, Int16, Int32, Int64, and Address.
For float and double, call stack shows it eventually calls putInt/putLong, and the exception comes from putInt/putLong.
Exception in thread "main" java.lang.InternalError
        at sun.misc.Unsafe.putLong(Native Method)
        at java.nio.DirectByteBuffer.putDouble(DirectByteBuffer.java:1030)
        at java.nio.DirectByteBuffer.putDouble(DirectByteBuffer.java:1053)
        at TTest.put(TTest.java:36)
        at TTest.main(TTest.java:23)